### PR TITLE
refactor test data

### PR DESCRIPTION
**Pull request by andiwand**

### CODE REVIEW

File: 3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -86,11 +91,28 @@ if (NOT json_POPULATED)
...
-    set(gtest_build_tests OFF CACHE BOOL "" FORCE)
-    set(gtest_build_samples OFF CACHE BOOL "" FORCE)
-    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
+
+if (ODR_TEST)
+    FetchContent_Populate(
+            odr-test-public
+            GIT_REPOSITORY https://github.com/opendocument-app/OpenDocument.test.git
+            GIT_TAG 58b5903
+            SOURCE_DIR ../test/input/odr-test-public
```
Comment (TomTasche) [inline 3rdparty/CMakeLists.txt:100]: is that where it's checked out? add to gitignore?

File: access/include/access/Path.h [hunk 2]
```diff
@@ -42,10 +42,10 @@ class Path final {
 
   std::string basename() const noexcept;
   std::string extension() const noexcept;
-  std::string fullExtension() const noexcept;
 
   Path parent() const;
   Path join(const Path &) const;
+  Path rebase(const Path &on) const;
```
Comment (TomTasche) [inline access/include/access/Path.h:48]: git 😂

File: odr/src/Meta.cpp [hunk 3]
```diff
@@ -1,39 +1,68 @@
...
     return "xls";
   default:
     return "unnamed";
   }
 }
 } // namespace
 
-namespace odr {
+FileType FileMeta::typeByExtension(const std::string &extension) noexcept {
+  if (extension == "zip")
```
Comment (TomTasche) [inline odr/src/Meta.cpp:39]: case like in typeToString looks cleaner in my opinion

File: test/DataDrivenTests.cpp [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+  Param(std::string input, const FileType type, const bool encrypted,
+        std::string password, std::string metaOutput, std::string htmlOutput)
+      : input{std::move(input)}, type{type}, encrypted{encrypted},
+        password{std::move(password)}, metaOutput{std::move(metaOutput)},
+        htmlOutput{std::move(htmlOutput)} {}
+};
+
+using Params = std::vector<Param>;
+
+class DataDrivenTest : public testing::TestWithParam<Param> {};
```
Comment (TomTasche) [inline test/DataDrivenTests.cpp:33]: is "datadriven" really the correct term here?

File: odr/src/Meta.cpp (hunk 3 above)
Comment (andiwand) [reply odr/src/Meta.cpp:39]: you can not switch case on strings

File: test/DataDrivenTests.cpp (hunk 4 above)
Comment (andiwand) [reply test/DataDrivenTests.cpp:33]: 🤷

File: 3rdparty/CMakeLists.txt (hunk 1 above)
Comment (andiwand) [reply 3rdparty/CMakeLists.txt:100]: it is in the build tree which is in gitignore anyway

**Pull request merged**